### PR TITLE
Update to purescript 0.11

### PR DESCRIPTION
**Pull request by wires**
Hi, FWIW I made some trivial changes to get this to build again with 0.11

- Changed `*` to `Type` and `!` to `Effect`.
- Updated dependencies to 3.0.0

### CODE REVIEW

File: src/Node/Buffer.purs [hunk 1]
```diff
@@ -23,7 +23,8 @@ module Node.Buffer
   ) where
 
 import Prelude
-import Control.Monad.Eff (Eff)
+import Prim (kind Type)
```
Comment (paf31) [inline src/Node/Buffer.purs]: You don't need to import this.

**Comment by wires** [comment]
Hmm, the build seems to fails on [unrecognised arguments](https://travis-ci.org/purescript-node/purescript-node-buffer#L834): `--censor-lib, --strict`. Do I need to insert `--` somewhere in the pulp call in `package.json`, or?

File: src/Node/Buffer.purs (hunk 1 above)
Comment (wires) [inline src/Node/Buffer.purs]: Ah, thanks! removed it

File: package.json [hunk 2]
```diff
@@ -5,9 +5,9 @@
     "build": "pulp build --censor-lib --strict"
```
Comment (paf31) [inline package.json]: You need to separate arguments with `--` now:

```
pulp build -- --censor-lib --strict
```

**Comment by jacereda** [comment]
`test/Main.purs` also requires changes for 0.11.1

File: package.json (hunk 2 above)
Comment (wires) [inline package.json]: Hum.., tried this before but it didn't work, pushed anyway but build is still failing.

I'm having problems building this locally, seems to be related to `pulp` (FWIW, I don't like this build tool, using my own Makefiles for my projects).

**Comment by paf31** [comment]
Thanks, but closing in favor of #19

**Pull request closed without merging**